aviary/cgcnn/data.py

```diff
@@ -211,103 +211,70 @@ def __getitem__(self, idx: int):
-        """
-        if dmin >= dmax:
-            raise AssertionError(
-                "Max radii must be larger than minimum radii for Gaussian basis expansion"
-            )
-        if dmax - dmin <= step:
-            raise AssertionError(
-                "Max radii below minimum radii + step size - please increase dmax."
-            )
-
-        self.filter = np.arange(dmin, dmax + step, step)
-        self.embedding_size = len(self.filter)
-
-        if var is None:
-            var = step
-
-        self.var = var
-
-    def expand(self, distances: np.ndarray) -> np.ndarray:
-        """Apply Gaussian distance filter to a numpy distance array
-
-        Args:
-            distances (ArrayLike): A distance matrix of any shape.
-
-        Returns:
-            np.ndarray: Expanded distance matrix with the last dimension of length len(self.filter)
-        """
-        distances = np.array(distances)
-
-        return np.exp(
-            -((distances[..., np.newaxis] - self.filter) ** 2) / self.var**2
-        )
-
-
 def collate_batch(
-    dataset_list: list[
-        tuple[Tensor, Tensor, LongTensor, Tensor | LongTensor, str | int]
+    dataset_list: tuple[
+        tuple[Tensor, Tensor, LongTensor, LongTensor],
+        list[Tensor | LongTensor],
+        list[str | int],
     ],
-) -> tuple[Tensor, Tensor, LongTensor, LongTensor, Tensor, list[str | int]]:
+) -> tuple[
+    Any,
+    ...
+    # tuple[Tensor, Tensor, Tensor, LongTensor, LongTensor, LongTensor, LongTensor],
+    # tuple[Tensor | LongTensor],
+    # tuple[str | int],
+]:
     """Collate a list of data and return a batch for predicting crystal properties.
 
     Args:
         dataset_list (list[tuple]): for each data point: (atom_fea, nbr_dist, nbr_idx, target)
-            - atom_fea (Tensor):
+            - atom_fea (Tensor): _description_
             - nbr_dist (Tensor):
+            - self_idx (LongTensor):
             - nbr_idx (LongTensor):
             - target (Tensor | LongTensor): target values containing floats for regression or
                 integers as class labels for classification
             - cif_id: str or int
 
     Returns:
-        tuple: containing
-        - batch_atom_fea (Tensor): Atom features from atom type
-        - batch_nbr_dist (Tensor): Bond features of each atom's M neighbors
-        - batch_nbr_idx (LongTensor): Indices of M neighbors of each atom
-        - crystal_atom_idx (list[LongTensor]): Mapping from the crystal idx to atom idx
-        - target (Tensor): Target value for prediction
-        - batch_cif_ids: list[str | int]: Identifiers like material_id, composition
+        tuple[
+            tuple[Tensor, Tensor, LongTensor, LongTensor, LongTensor]: batched CGCNN model inputs,
+            tuple[Tensor | LongTensor]: Target values for different tasks,
+            # TODO this last tuple is unpacked how to do type hint?
+            *tuple[str | int]: Identifiers like material_id, composition
+        ]
```

Comment:
@janosh

Reply:
Didn't I type-hint this function before (in an admittedly poor manner)? I guess we can just go with a `tuple[Any, ...]` and call it done if you want sth shorter than what I wrote before.

Reply:
Oh wait, I was mixing this up with `collate_batch`. My bad.

Reply:
You did but asked me to try again but it doesn't return the list the list is unpacked. I think it's just hard to do the hints. Can try again in next PR?

Reply:
Not that important, I'd say. The function isn't even user-facing so it's mostly for our benefit.